lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
```

Comment:
`retriable` is honestly a pretty generic thing to be injecting into `Kernel` - no more or less generic, at least, than `with_context` IMHO